tests/spec/fs.unlink.spec.js

```diff
@@ -104,3 +104,37 @@ describe('fs.unlink', function() {
     });
   });
 });
+
+/**
+ * fsPromises tests
+ */
+
+describe('fsPromises.unlink', function() {
+  beforeEach(util.setup);
+  afterEach(util.cleanup);
+
+  it('should not unlink directories', () => {
+    var fs = util.fs().promises;
+
+    return fs.mkdir('/mydir')
+      .then(() => {
+        return fs.unlink('/mydir')
+          .catch(error => {
+            expect(error).to.exist;
+            expect(error.code).to.equal('EPERM');
+          });
+      })
+      .then(() => {
+        return fs.stat('/mydir')
+          .then(stats => {
+            expect(stats).to.exist;
+            expect(stats.type).to.equal('DIRECTORY');
+          })
+          .then( error => {
```

Comment:
1. This `.then` block is not needed. This is an error that may happen, but not part of the actual test (for example, in the case of `fs.unlink('/mydir')` (line 121), for the test to pass an error must exist. Which is why you used a `.catch`). If an error occurred during `fs.stat('/mydir')` (line 128), it'll be caught in the `.catch` on line 137. Once that happens the test will fail anyway, so you don't need to explicitly test for an error to not exist.

2. Consider the following block:
```js
	return fs.stat('/mydir')
   		.then(stats => {...})
		.then( error => {..})
```
The first argument of `.then` is the return value if the promise was **successfully** resolved. `error` in `.then( error => {..})` would represent the return value of the `.then` block before it if it was a promise, and that promise was successfully resolved. However, the `.then` block (line 128) contains chai assertions. Therefore, `.then( error => {..})` part is not needed.